### PR TITLE
feat: add tax indicator input validation

### PR DESCRIPTION
**Pull request by Mech0z**
## Description

Ensure a charge information request have Tax Indicator set to a valid value.

## References

* #1326

### CODE REVIEW

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Domain/Dtos/ChargeCommands/Validation/InputValidation/Factories/ChargeOperationInputValidationRulesFactory.cs [hunk 1]
```diff
@@ -97,6 +97,7 @@ private List<IValidationRuleContainer> CreateRulesForChargeInformation(ChargeOpe
                 CreateRuleContainer(new ChargeDescriptionRequiredRule(chargeOperationDto), chargeOperationDto.Id),
                 CreateRuleContainer(new ResolutionIsRequiredRule(chargeOperationDto), chargeOperationDto.Id),
                 CreateRuleContainer(new TransparentInvoicingValidationRule(chargeOperationDto), chargeOperationDto.Id),
+                CreateRuleContainer(new TaxIndicatorValidationRule(chargeOperationDto), chargeOperationDto.Id),
```
Comment (prtandrup) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Domain/Dtos/ChargeCommands/Validation/InputValidation/Factories/ChargeOperationInputValidationRulesFactory.cs:100]: Should we do this, like `ResolutionIsRequiredRule`:

```suggestion
                CreateRuleContainer(new TransparentInvoicingIsRequiredValidationRule(chargeOperationDto), chargeOperationDto.Id),
                CreateRuleContainer(new TaxIndicatorIsRequiredValidationRule(chargeOperationDto), chargeOperationDto.Id),
```

Also, it appears that some of our rules are called "ValidationRule" while other are just "Rule". Maybe we should align, perhaps in another PR.

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Infrastructure.Core/Cim/ValidationErrors/CimValidationErrorTextTemplateMessages.cs [hunk 2]
```diff
@@ -177,6 +177,10 @@ public static class CimValidationErrorTextTemplateMessages
         public const string TransparentInvoicingIsRequiredErrorText =
             "Transparent invoicing is missing.";
 
+        [ErrorMessageFor(ValidationRuleIdentifier.TaxIndicatorIsRequired)]
+        public const string TaxIndicatorIsRequiredErrorText =
+            "Tax indicator invoicing is missing.";
```
Comment (prtandrup) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Infrastructure.Core/Cim/ValidationErrors/CimValidationErrorTextTemplateMessages.cs:182]:
```suggestion
            "Tax indicator is missing.";
```

**Review by prtandrup — APPROVED** [review]

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Domain/Dtos/ChargeCommands/Validation/InputValidation/Factories/ChargeOperationInputValidationRulesFactory.cs (hunk 1 above)
Comment (Mech0z) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Domain/Dtos/ChargeCommands/Validation/InputValidation/Factories/ChargeOperationInputValidationRulesFactory.cs]: done

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/Domain/Dtos/ChargeCommands/Validation/InputValidation/ValidationRules/TaxIndicatorValidationRuleTests.cs [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+using GreenEnergyHub.Charges.Domain.Dtos.ChargeCommands.Validation.InputValidation.ValidationRules;
+using GreenEnergyHub.Charges.TestCore.Attributes;
+using GreenEnergyHub.Charges.Tests.Builders.Command;
+using Xunit;
+using Xunit.Categories;
+
+namespace GreenEnergyHub.Charges.Tests.Domain.Dtos.ChargeCommands.Validation.InputValidation.ValidationRules
+{
+    [UnitTest]
+    public class TaxIndicatorValidationRuleTests
```
[the remark this answers is not in the document]
Comment (Mech0z) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/Domain/Dtos/ChargeCommands/Validation/InputValidation/ValidationRules/TaxIndicatorValidationRuleTests.cs]: done

**Pull request merged**